assimilation_code/modules/assimilation/algorithm_info_mod.f90

```diff
@@ -128,107 +290,198 @@ subroutine probit_dist_info(kind, is_state, is_inflation, dist_type, &
 ! In the long run, may not have to have separate controls for each of the input possibilities
 ! However, for now these are things that need to be explored for science understanding
 
-if(is_inflation) then
+! Fill arrays with possible dist_type strings and corresponding ints
+possible_dist_types(1) = 'NORMAL_DISTRIBUTION'
+possible_dist_types(2) = 'BOUNDED_NORMAL_RH_DISTRIBUTION'
+possible_dist_types(3) = 'GAMMA_DISTRIBUTION'
+possible_dist_types(4) = 'BETA_DISTRIBUTION'
+possible_dist_types(5) = 'LOG_NORMAL_DISTRIBUTION'
+possible_dist_types(6) = 'UNIFORM_DISTRIBUTION'
+possible_dist_types(7) = 'PARTICLE_FILTER_DISTRIBUTION'
+
+possible_dist_type_ints(1) = 1
+possible_dist_type_ints(2) = 2
+possible_dist_type_ints(3) = 3
+possible_dist_type_ints(4) = 4
+possible_dist_type_ints(5) = 5
+possible_dist_type_ints(6) = 6
+possible_dist_type_ints(7) = 7
```

Comment:
This is a definite change.

The parameters are from distribution_params_mod
```
use distribution_params_mod, only : NORMAL_DISTRIBUTION, BOUNDED_NORMAL_RH_DISTRIBUTION, &
   GAMMA_DISTRIBUTION, BETA_DISTRIBUTION, LOG_NORMAL_DISTRIBUTION, UNIFORM_DISTRIBUTION,  &
   PARTICLE_FILTER_DISTRIBUTION
````
so we don't want to be redefining these parameters here (ints 1-7). This will cause bugs when someone changes distribution_params_mod.

The various types should store the integer value not the character value:
```
type probit_inflation_type
-  character(len=129)    :: dist_type
+ integer              :: dist_type
   logical               :: bounded_below, bounded_above
   real(r8)              :: lower_bound,   upper_bound
end type
```

Do the conversion from string to integer value when you fill in the type.